File: WEEK_3_Python/Q1.py
class Node:
    def __init__(self, data):
        self.data, self.next = data, None

def print_list(head):
    cur = head
    while cur:
        print(cur.data, end=" ")
        cur = cur.next
    print()

def insert_at_begin(head, val):
    node = Node(val)
    node.next = head
    return node

# Example
head = None
for x in [3, 2, 1]:
    head = insert_at_begin(head, x)
print_list(head)   

File: WEEK_3_C/Q1.c
#include <stdio.h>
#include <stdlib.h>

struct Node { int data; struct Node *next; };

void print(struct Node* head){
    while(head){ printf("%d ",head->data); head=head->next; }
    printf("\n");
}

int main(){
    struct Node *head=NULL,*t;
    for(int i=3;i>=1;i--){
        t=(struct Node*)malloc(sizeof(struct Node));
        t->data=i; t->next=head; head=t;
    }
    print(head);
    return 0;
}